### PR TITLE
[luci-interpreter] Support BatchMatMul Op

### PR DESCRIPTION
**Pull request by jinevening**
This supports BatchMatMul Operator in luci-interpreter.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Draft PR: #8334 
Related to: #8325

### CODE REVIEW

File: compiler/luci-interpreter/pal/linux/PALBatchMatMul.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+{
+inline void BatchMatMul(const tflite::RuntimeShape &lhs_shape, const float *lhs_data,
+                        const tflite::RuntimeShape &rhs_shape, const float *rhs_data,
+                        const tflite::RuntimeShape &output_shape, float *output_data)
+{
+  tflite::reference_ops::BatchMatMul(lhs_shape, lhs_data, rhs_shape, rhs_data, output_shape,
+                                     output_data);
+}
+
+static inline void SetupScratchpadTensor(luci_interpreter::Tensor *lhs_scratchpad,
```
Comment (jinevening) [inline compiler/luci-interpreter/pal/linux/PALBatchMatMul.h:32]: For reviewers: BatchMatMul has two scratchpad memory: one for transposed lhs and another for transposed rhs.

File: compiler/luci-interpreter/src/loader/nodes/BatchMatMul.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  lhs_scratchpad->set_data_buffer(nullptr);
+  auto rhs_scratchpad =
+    std::make_unique<Tensor>(rhs->element_type(), Shape({}), AffineQuantization{}, "");
+  rhs_scratchpad->set_observable(false);
+  rhs_scratchpad->set_data_buffer(nullptr);
+  // If node has execution plan then read memory offsets for scratchpad temporary tensor
+  // from the beginning of shared memory buffer.
+  // Used in Static Memory Manager.
+  // TODO move tensors offset initialization to one place
+  if (luci::has_execution_plan(node))
```
Comment (jinevening) [inline compiler/luci-interpreter/src/loader/nodes/BatchMatMul.cpp:49]: @binarman 

Could you please look at this code? I'm not sure this code will work well with static memory manager.

File: compiler/luci-interpreter/src/kernels/BatchMatMul.test.cpp [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  _memory_manager->allocate_memory(rhs_scratch);
+  _memory_manager->allocate_memory(output_tensor);
+  kernel.execute();
+
+  EXPECT_THAT(extractTensorData<float>(output_tensor),
+              FloatArrayNear({427., 448., 469., 490., 1039., 1096., 1153., 1210.}));
+  EXPECT_THAT(extractTensorShape(output_tensor), ::testing::ElementsAreArray({2, 1, 4}));
+}
+
+TEST_F(BatchMatMulTest, Invalid_Shape_NEG)
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/kernels/BatchMatMul.test.cpp:181]: optional but if possible, more negative tests will be great...

File: compiler/luci-interpreter/src/loader/nodes/BatchMatMul.cpp (hunk 2 above)
Comment (binarman) [reply compiler/luci-interpreter/src/loader/nodes/BatchMatMul.cpp:49]: You did everything right, this is how we manage scratchpad tensors :+1:

**Review by binarman — DISMISSED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by binarman — APPROVED** [review]

**Pull request merged**